assets/admin/editor-wizard/editor-wizard-modal.js

```diff
@@ -3,83 +3,29 @@
  */
 import { useDispatch } from '@wordpress/data';
 import { Modal } from '@wordpress/components';
-import { useEffect, useLayoutEffect, useState } from '@wordpress/element';
-import { store as blockEditorStore } from '@wordpress/block-editor';
+import { useState } from '@wordpress/element';
 import { store as editorStore } from '@wordpress/editor';
 
 /**
  * Internal dependencies
  */
 import Wizard from './wizard';
 import useEditorWizardSteps from './use-editor-wizard-steps';
+import { useWizardOpenState, useDefaultTemplate } from './helpers';
 import '../../shared/data/api-fetch-preloaded-once';
 
-/**
- * A React Hook to observe if a modal is open based on the body class.
- *
- * @param {boolean} shouldObserve If it should observe the changes.
- *
- * @return {boolean|undefined} Whether a modal is open, or `undefined` if it's not initialized yet.
- */
-const useObserveOpenModal = ( shouldObserve ) => {
-	const [ hasOpenModal, setHasOpenModal ] = useState();
-
-	useEffect( () => {
-		if ( ! shouldObserve ) {
-			return;
-		}
-
-		// Initialize state after modals are open or not.
-		setTimeout( () => {
-			setHasOpenModal( document.body.classList.contains( 'modal-open' ) );
-		}, 1 );
-
-		const observer = new window.MutationObserver( () => {
-			setHasOpenModal( document.body.classList.contains( 'modal-open' ) );
-		} );
-		observer.observe( document.body, {
-			attributes: true,
-			attributeFilter: [ 'class' ],
-		} );
-
-		return () => {
-			observer.disconnect();
-		};
-	}, [ shouldObserve ] );
-
-	return hasOpenModal;
-};
-
-/**
- * A React Hook to control the wizard open state.
- *
- * @return {boolean} Whether the modal should be open.
- */
-const useWizardOpenState = () => {
-	const [ open, setOpen ] = useState( false );
-	const [ done, setDone ] = useState( false );
-	const hasOpenModal = useObserveOpenModal( ! done );
-
-	useLayoutEffect( () => {
-		if ( done ) {
-			setOpen( false );
-		} else if ( false === hasOpenModal ) {
-			// If no modal is open, it's time to open.
-			setOpen( true );
-		}
-	}, [ done, hasOpenModal ] );
-
-	return [ open, setDone ];
-};
-
 /**
  * Editor wizard modal component.
  */
 const EditorWizardModal = () => {
-	const [ open, setDone ] = useWizardOpenState();
-	const { synchronizeTemplate } = useDispatch( blockEditorStore );
+	const dataState = useState( {} );
```

Comment:
It lifts up the data state from the wizard in order to have it available when skipping the wizard. We need the description to add to the post.